include/vcpkg/base/parallel-algorithms.h

```diff
@@ -29,55 +30,55 @@ namespace vcpkg
         }
     }
 
-    template<class RanIt, class F>
-    void parallel_for_each_n(RanIt begin, size_t work_count, F cb) noexcept
+    template<class T, class F>
+    void parallel_for_each(View<T> view, F cb) noexcept
     {
-        if (work_count == 0)
+        if (view.size() == 0)
         {
             return;
         }
-        if (work_count == 1)
+        if (view.size() == 1)
         {
-            cb(*begin);
+            cb(view[0]);
             return;
         }
 
         std::atomic_size_t next{0};
 
-        execute_in_parallel(work_count, [&]() {
+        execute_in_parallel(view.size(), [&]() {
             size_t i = 0;
-            while (i < work_count)
+            while (i < view.size())
             {
                 if (next.compare_exchange_weak(i, i + 1, std::memory_order_relaxed))
                 {
-                    cb(*(begin + i));
+                    cb(view[i]);
                 }
             }
         });
     }
 
-    template<class RanItSource, class RanItTarget, class F>
-    void parallel_transform(RanItSource begin, size_t work_count, RanItTarget out_begin, F&& cb) noexcept
+    template<class T, class RanItTarget, class F>
```

Comment:
Can you also add the following comment because this behavior differs from std::transform?
`PRE: out_begin must point to a range that is at least as large as the input range` I forgot to add this.